### DISCUSSION TITLE
Adobe Ai SVG is scaled down in LaserGRBL

### DISCUSSION THREAD

**Discussion by Orangbelanda** [arkypita/LaserGRBL#1495] [Q&A]
The vector graphics in LaserGRBL is about 25% smaller than the original SVG file from Adobe Illustrator. No idea why this happens, but it happens all the time.

**Comment by Orangbelanda** [comment]
[100x100onA4.zip](https://github.com/arkypita/LaserGRBL/files/5909656/100x100onA4.zip) The Ai and SVG files as example.

**Comment by arkypita** [comment]
If I open it in Inkscape it show 75*75 rectangle in a non A4 page size

![image](https://user-images.githubusercontent.com/8782035/106574904-c22ae900-653b-11eb-9e1e-263258a52f0f.png)

![image](https://user-images.githubusercontent.com/8782035/106575006-e090e480-653b-11eb-9eab-05e6343cd944.png)

**Comment by svenhb** [comment]
As I know: the standard is 96 dpi (used in inkscape) but AI  may save with 72 dpi.
https://community.adobe.com/t5/illustrator/illustrator-svg-export-dimension-problem/td-p/10159751

**Comment by Orangbelanda** [comment]
Yes ! This must be the cause. If I scale up the Ai file to 133,333 percent, the dimensions saved to SVG are exactly right. Its a bit weird  work around, but it works :-) Would be nice if LaserGRBL could do this for me (while import Ai SVG).

**Comment by svenhb** [comment]
I would say it is a problem of AI, since 96 dpi is the standard :-)

**Comment by Orangbelanda** [comment]
OK, OK, OK, I am not defending Ai ;-)

**Comment by arkypita** [comment]
SVG support in LaserGRBL is in a rather embryonic state. The main problem is that there is no single standard of SVG, and that there are dozens of possible transformation / scaling operations, units of measurement, viewports etc, which different programs implement in different ways, which is a headache to manage them all.

Moreover, it often happens that if you touch something for one then it no longer works for another.

The other problem is that the code I use to read the SVG I didn't write it myself, but I got it from another third party project, so I don't have 100% knowledge of what it does, and how to change it.

It is for these reasons that SVG support is so poor at present in LaserGRBL.

In my tests I preferred to align myself with the SVG of inkscape, as being a free editor it can be used by everyone.

**Comment by Orangbelanda** [comment]
I understand such headaches, not as a programmer, but as an industrial designer. And I appreciate your choice for Inkscape compatibility. How about EPS file format, will that give such headaches as well ?

**Comment by Orangbelanda** [comment]
Strange, I tried an other 133,333 percent scaled SVG file from Ai but the resulting scale in LaserGRBL was wrong. Anyway, I decided to use Inkscape to open the original Ai file, and than save it to SVG (Inkscape version), which works very well.

**Comment by sanyats** [comment]
у меня при работе с SVG файлами LaserGRBL уходит в критическую ошибку, что может быть, как исправить?

**Comment by Orangbelanda** [comment]
Файл SVG должен создаваться Inkscape, поскольку разные программы создают разные версии SVG. Хочешь верь, хочешь нет ;-)
